eng/scripts/Language-Settings.ps1

```diff
@@ -1,82 +1,127 @@
-$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+$LanguageDisplayName = "go"
 
-# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
-function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+# get version from specific files (*constants.go, *version.go)
+function Get-GoModuleVersionInfo ($modPath)
 {
-    $workFolder = $pkg.Directory
-    $releaseNotes = ""
+  $VERSION_LINE_REGEX = ".+\s*=\s*`".*v(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
+
+  $versionFiles = Get-ChildItem -Recurse -Path $modPath -Filter *.go
+
+  # for each version file, use regex to search go version num
+  foreach ($versionFile in $versionFiles)
+  {
+    # limit the search to constant and version file
+    if (!$versionFile.Name.Contains("constant") -and !$versionFile.Name.Contains("version")) {
+      continue
+    }
+    $content = Get-Content $versionFile -Raw
+    # finding where the version number are
+    if ($content -match $VERSION_LINE_REGEX) {
+        return "$($matches["version"])", $versionFile
+    }
+  }
+
+  LogWarning "Unable to find version for $modPath"
+  return $null
+}
 
-    if ($workFolder -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
+function Get-GoModuleProperties($goModPath)
+{
+    if ($goModPath -match "(?<modPath>sdk[\\/](?:(?<modGroup>[^\\/]+)[\\/])?(?<modName>[^\\/]+$))")
     {
-        if ($matches["arm"])
-        {
-            $packageName = "arm" + $matches["pkgname"]
-            $rpName = $matches["service"]
-            $pkgId = "sdk/$rpName/$packageName"
-        }
-        else
-        {
-            $packageName = $matches["pkgname"]
-            $pkgId = "sdk/$packageName"
-        }
+      $modPath = $matches["modPath"] -replace "\\", "/"
+      $modName = $matches["modName"] # We may need to start readong this from the go.mod file if the path and mod config start to differ
+      $modGroup = $matches["modGroup"]
+      $sdkType = "client"
+      if ($modName.StartsWith("arm")) { $sdkType = "mgmt" }
+
+      $modVersion, $versionFile = Get-GoModuleVersionInfo $goModPath
+
+      if (!$modVersion) {
+        return $null
+      }
+
+      $pkgProp = [PackageProps]::new($modPath, $modVersion, $goModPath, $modGroup)
+      $pkgProp.IsNewSdk = $true
+      $pkgProp.SdkType = $sdkType
+
+      $pkgProp | Add-Member -NotePropertyName "VersionFile" -NotePropertyValue $versionFile
+      $pkgProp | Add-Member -NotePropertyName "ModuleName" -NotePropertyValue $modName
+
+      return $pkgProp
     }
+    return $null
+}
 
-    $pkgVersion = Get-Version $workFolder
+# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
+function Get-go-PackageInfoFromPackageFile($pkg, $workingDirectory)
+{
+    $workFolder = $pkg.Directory
```

Comment:
Am not sure that waht is passed here is the DirectoryName?

Reply:
Fixed.